From 9fff30285f5bc90bf0932864e86b9939b04c6c51 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 05 Oct 2021 13:36:18 +0300
Subject: [PATCH] [SW-57909] sec: remove large data structs from stack

Large data structures were places on the stack and caused stack
overflow when secrity is enabled.
Moving those data structs from stack to bss section.

Change-Id: I29d0a243f2fbabeb7fb53a077a4267a199786e50
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index bb68623..2838338 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -2153,13 +2153,14 @@
 	boot_info_set(HL_BOOT_INFO_PUB_KEY_HASH, key, len);
 }
 
+static struct hl_fit_sec_desc fit_sec;
+static u8 signature[FIT_SIGNATURE_SIZE];
+
 int preboot_verify_img_sig(const void *img, size_t img_size,
 			   enum fit_sec_failure *res, u32 *key_idx)
 {
-	struct hl_fit_sec_desc fit_sec;
 	const void *fit_sec_off;
 	const void *pub_key_fit;
-	u8 signature[FIT_SIGNATURE_SIZE];
 
 	if (!preboot_security_is_enabled())
 		return 0;
